File: 2021/2021W31/PreppindataW31.py
import pandas as pd

sales = pd.read_csv("F:\\Data\\PD 2021 Wk 31 Input.csv")
sales = sales[sales['Status'] != 'Return to Manufacturer'][['Store', 'Item', 'Number of Items']].copy() 
sales['Item sold per store'] = sales['Number of Items'].groupby(sales['Store']).transform('sum')
sales = sales.pivot_table(index=['Store', 'Item sold per store'], columns='Item', values='Number of Items', aggfunc=sum).reset_index()
sales = sales[['Store', 'Wheels', 'Tyres', 'Saddles', 'Brakes', 'Item sold per store']]

File: 2021/2021W31/preppindataW31.R
library(readr)
library(dplyr)
library(tidyr)

final <- read_csv("F:/Data/PD 2021 Wk 31 Input.csv") %>%
  filter(`Status` != 'Return to Manufacturer') %>%
  select(c('Store', 'Item', 'Number of Items')) %>%
  group_by(`Store`) %>%
  mutate('Item sold per store' = sum(`Number of Items`)) %>%
  pivot_wider(id_cols=c('Store', 'Item sold per store'),  names_from='Item', values_from='Number of Items', values_fn = sum) %>%
  select(c('Store', 'Wheels', 'Tyres', 'Saddles', 'Brakes', 'Item sold per store'))

View(final)
